core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -959,7 +959,10 @@ public synchronized void setInitialTopologyId(int initialTopologyId) {
    public boolean resolveConflictsOnMerge() {
       // It doesn't make sense to resolve conflicts if we are not going to rebalance the cache as entries on "old" owners
       // will not be deleted when no rebalance occurs.
-      return resolveConflictsOnMerge && clusterTopologyManager.isRebalancingEnabled() && rebalancingEnabled;
+      return resolveConflictsOnMerge &&
+            clusterTopologyManager.getManagerStatus().isRunning() &&
```

Comment:
The CTMI status only changes when CTMI itself is stopping, but the caches are all stopped before this point. IIRC conflict resolution requires the cache to be running on the coordinator, so you'd need to check for that explicitly.